src/leappto/cli.py

```diff
@@ -773,28 +772,40 @@ def set_port(self, protocol, source, target = None):
 
 
 
-def _port_scan(ip, port_range = None, shallow = False):
-    scan_args = '-sS' if shallow else '-sV'
+def _port_scan(ip, port_range=None, shallow=False, force_nmap=False):
 
-    port_scanner = nmap.PortScanner()
-    port_scanner.scan(ip, port_range, arguments=scan_args)
-    scan_info = port_scanner.scaninfo()
+    def _nmap(port_list, ip, port_range=None, shallow=False):
+        if shallow and port_range is None:
+            port_range = '{}-{}'.format(_MIN_PORT, _MAX_PORT)
+        scan_args = '-sS' if shallow else '-sV'
 
-    if scan_info.get('error', False):
-        raise PortScanException(scan_info['error'][0] if isinstance(scan_info['error'], list) else scan_info['error'])
-    elif ip not in port_scanner.all_hosts():
-        raise PortScanException("Machine {} not found".format(ip))
+        port_scanner = nmap.PortScanner()
+        port_scanner.scan(ip, port_range, scan_args)
+        scan_info = port_scanner.scaninfo()
 
-    ports = PortList()
+        if scan_info.get('error', False):
+            raise PortScanException(
+                scan_info['error'][0] if isinstance(scan_info['error'], list) else scan_info['error']
+            )
 
-    for proto in port_scanner[ip].all_protocols():
-        for port in sorted(port_scanner[ip][proto]):
-            if port_scanner[ip][proto][port]['state'] != 'open':
-                continue
+        for proto in port_scanner[ip].all_protocols():
+            for port in sorted(port_scanner[ip][proto]):
+                if port_scanner[ip][proto][port]['state'] in ('open', 'filtered'):
+                    port_list.set_port(proto, port, port_scanner[ip][proto][port])
+        return port_list
 
-            ports.set_port(proto, port, port_scanner[ip][proto][port])
+    def _net_util(port_list):
+        sconns = psutil.net_connections(kind=port_list.PROTO_TCP)
+        for sconn in sconns:
+            addr, port = sconn.laddr
+            if not port_list.has_port(port_list.PROTO_TCP, port):
```

Comment:
Port mapping algorithm doesn't support ports with various ips, that's why I check if port is open on the local machine (I don't care where it's assigned, docker will bind port to ::: as default, so have to check everything). I will create a card which will cover above feature.